packages/core/src/engine/Autodiff.ts

```diff
@@ -1755,21 +1903,25 @@ const genBytes = (graphs: ad.Graph[]): Uint8Array => {
  * @param graphs an array of graphs to compile
  * @returns a compiled/instantiated WebAssembly function
  */
-export const genCode = async (...graphs: ad.Graph[]): Promise<Gradient> =>
-  await Gradient.make(
+export const genCode = async (...graphs: ad.Graph[]): Promise<Gradient> => {
+  const meta = makeMeta(graphs);
```

Comment:
My understanding is this allocates WASM memory in the autodiff module (not `@penrose/optimizer`) with built-in functions, which is then imported by the compiled big autodiff comp graph function.

Reply:
Correct, the `makeMeta` function creates a Wasm memory big enough to hold all the memory we use for the generated computation function; then when we instantiate our generated Wasm module, we pass it that memory object.